app/jobs/create_new_fake_sandbox_data_job.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class CreateNewFakeSandboxDataJob < CronJob
+  self.cron_expression = "0 2 * * *"
+
+  def perform
+    return unless Rails.env.sandbox? || Rails.env.test?
+
+    provider_name = "Education Development Trust"
+
+    edt_ecf_lead_provider = LeadProvider.find_by(name: provider_name)
+    edt_npq_lead_provider = NPQLeadProvider.find_by(name: provider_name)
+
+    school = edt_ecf_lead_provider.schools.first
+    school_cohort = SchoolCohort.find_by(school: school, cohort: Cohort.current)
```

Comment:
Maybe you can extract `edt_ecf_lead_provider`, `edt_nqp_lead_provider`, `school` and `school_cohort` in private methods?